src/plugins/data/public/search/expressions/eql.test.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { getEql } from './eql';
+import { MockedKeys } from '@kbn/utility-types/jest';
+import { EqlExpressionFunctionDefinition } from '../../../common/search/expressions';
+import { StartServicesAccessor } from 'kibana/public';
+import { DataPublicPluginStart, DataStartDependencies } from '../../types';
+import { of } from 'rxjs';
+
+jest.mock('@kbn/i18n', () => {
+  return {
+    i18n: {
+      translate: (id: string, { defaultMessage }: { defaultMessage: string }) => defaultMessage,
+    },
+  };
+});
+
+describe('eql', () => {
+  let getStartServices: StartServicesAccessor<DataStartDependencies, DataPublicPluginStart>;
+  let startDependencies: MockedKeys<
+    StartServicesAccessor<DataStartDependencies, DataPublicPluginStart>
+  >;
+  let eql: EqlExpressionFunctionDefinition;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    startDependencies = [
+      {
+        uiSettings: {
+          get: jest.fn().mockReturnValue(true),
+        },
+      },
+      {},
+      {
+        search: {
+          search: jest.fn((params: any) => of({ rawResponse: { body: params } })),
+        },
+      },
+    ];
+    getStartServices = jest
+      .fn()
+      .mockResolvedValue(new Promise((resolve) => resolve(startDependencies)));
+    eql = getEql({ getStartServices });
+  });
+
+  describe('correctly handles input', () => {
```

Comment:
This could be simplified with `test.each`. That can also help to get rid of `as any`.